database/query/entities.sql

```diff
@@ -99,4 +99,13 @@ WHERE entity_id = $1;
 
 -- name: DeleteAllPropertiesForEntity :exec
 DELETE FROM properties
-WHERE entity_id = $1;
\ No newline at end of file
+WHERE entity_id = $1;
+
+-- name: GetTypedEntitiesByProperty :many
```

Comment:
This is not useful in high performance code like the webhook updates because there's no way to index the properties well.

I can see two ways here:
 1) Add the upstream ID as a string value to the properties table
 2) Don't even query by upstream ID from minder-core, but always by name. I'm not sure if all providers will give us this option.

Reply:
We actually can have a partial index (using a GIN index) that would help here. e.g.

```
CREATE INDEX ON properties((value->>'upstream_id'));
```

or something of the sort.